internals/models.py

```diff
@@ -938,8 +953,8 @@ def __notify_feature_subscribers_of_changes(self, is_update):
        posting to a task queue."""
     # Diff values to see what properties have changed.
     changed_props = []
-    for prop_name, prop in self.properties().iteritems():
-      if prop_name in ('created_by', 'updated_by'):
+    for prop_name, prop in self._properties.iteritems():
+      if prop_name in ('created_by', 'updated_by', 'updated', 'created'):
         continue
       new_val = getattr(self, prop_name, None)
       old_val = getattr(self, '_old_' + prop_name, None)
```

Comment:
The properties `updated` and `created` are essentially of type `datetime.datetime`. Hence, if ever (although impossible while using old DB Client Library) there is change between `_old_updated` and `updated`, the following error will be raised if changed in line 957 (above) are ignored.

>  TypeError: datetime.datetime(2021, 6, 9, 10, 46, 8, 833823) is not JSON serializable

Although the above case is impossible to happen if use DB Client Library, the same is not true for the not-so-old NDB Client Library. This is because the arguments `auto_add` and `auto_add_now` only assign the values once the entity is written to datastore. As this [link](https://cloud.google.com/appengine/docs/standard/python/ndb/entity-property-reference#Date_and_Time) says, 

>  The automatic value is not generated until the entity is written; that is, these options don't provide dynamic defaults. (These details differ from the old db API.)

Hence, in case of the not-so-old NDB Client Library, `_old_updated` and `updated` have the same value `None` initially but once the entity is written, the value of updated changes to the required value.

Reply:
OK. I think I understand.  Thanks for working through that.

Reply:
@jrobbins, can we mark this as resolved ?

Reply:
Yes.